.github/workflows/build-test.yml

```diff
@@ -0,0 +1,169 @@
+name: build-test
```

Comment:
Calling this `build-test` ports over the CircleCI name directly, but we already have a significant footprint of GHA tests so this name doesn't tell us why it's in its own workflow.

Reply:
Yeah I totally agree - there doesn't seem to be a clear reason as to why these tests are separated from the rest in `test-core`? Unsure if y'all want to keep the split or just combine them all into one workflow?

Reply:
Ah ok gotcha - makes sense. I've gone ahead and split these tests out by _function_ then to hopefully add more clarity:

1. e2e
2. ui
3. windows

**note the e2e includes the builds that are used by the `nomad-e2e` repo